src/blas/backends/portblas/portblas_common.hpp

```diff
@@ -199,7 +199,12 @@ struct throw_if_unsupported_by_device {
         auto fn = [](auto&&... targs) {                                                         \
             portBLASFunc(std::forward<decltype(targs)>(targs)...);                              \
         };                                                                                      \
-        std::apply(fn, args);                                                                   \
+        try {                                                                                   \
+            std::apply(fn, args);                                                               \
+        }                                                                                       \
+        catch (const std::runtime_error& e) {                                                   \
```

Comment:
I wonder if this is overly broad? I think any synchronous exceptions from the SYCL runtime are going to get rethrown as `unimplemented` which is misleading.

Reply:
My thoughts too, I think we should only catch `blas::unsupported_exception` once https://github.com/codeplaysoftware/portBLAS/pull/524 is merged.

Reply:
Thank you both for the review. I agree and I fixed in [6b259b8](https://github.com/oneapi-src/oneMKL/pull/525/commits/6b259b84e3e7603a31f061ef28a26f2d289f7a46)